src/Transformers/ContentTransformer.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Railroad\MusoraApi\Transformers;
+
+use League\Fractal\TransformerAbstract;
+use Railroad\Railcontent\Helpers\ContentHelper;
+use Railroad\Railcontent\Support\Collection;
+
+class ContentTransformer extends TransformerAbstract
+{
+    public function transform($content, $responseStructure = [])
+    {
+        $response = [];
+
+        if (empty($responseStructure) && array_key_exists('type', $content)) {
+
+            $type = (in_array($content['type'], config('railcontent.showTypes'))) ? 'show-lesson' : $content['type'];
+
+            $responseStructure = config('musora-api.response-structure.' . $type);
+
+        }
+
+        if (!$responseStructure) {
+            return (!is_array($content)) ? $content->getArrayCopy() : $content;
+        }
+
+        foreach ($responseStructure as $index => $value) {
```

Comment:
There may be an opportunity to refactor some of this type of logic among all the transformers. I also found it very hard to follow what's going on. Some explanation comments would be useful